flink-core/src/test/java/org/apache/flink/configuration/ConfigurationTest.java

```diff
@@ -403,4 +405,43 @@ public void testRemove(){
 		assertEquals("Wrong expectation about size", cfg.keySet().size(), 0);
 		assertFalse("Expected 'unexistedOption' is not removed", cfg.removeConfig(unexistedOption));
 	}
+
+	@Test
+	public void testShouldParseValidStringToEnum() {
+		final ConfigOption<String> validOption = ConfigOptions
+			.key("test-string-key")
+			.noDefaultValue();
+
+		final Configuration configuration = new Configuration();
+		configuration.setString(validOption.key(), TestEnum.VALUE1.toString());
+
+		final TestEnum parsedEnumValue = configuration.getAsEnum(TestEnum.class, validOption);
+		assertEquals(TestEnum.VALUE1, parsedEnumValue);
+	}
+
+	@Test
+	public void testThrowsExceptionIfTryingToParseInvalidStringForEnum() {
+		final ConfigOption<String> validOption = ConfigOptions
+			.key("test-string-key")
+			.noDefaultValue();
+
+		final Configuration configuration = new Configuration();
+		final String invalidValueForTestEnum = "InvalidValueForTestEnum";
+		configuration.setString(validOption.key(), invalidValueForTestEnum);
+
+		try {
+			configuration.getAsEnum(TestEnum.class, validOption);
+			fail("Expected exception not thrown");
+		} catch (IllegalArgumentException e) {
+			final String expectedMessage = "Value for config option " +
```

Comment:
nit: Personally prefer the approach with `@Rule`

```
	@Rule
	public ExpectedException thrown = ExpectedException.none();

    @Test
     public testMethod() {
       ...
       thrown.expect(IllegalArgumentException.class);
       thrown.message(....);
      }
```

Reply:
Current solution is legible, and frequently used in Flink and elsewhere. I'll leave it as is.